sql/sql_insert.cc

```diff
@@ -681,7 +681,8 @@ bool Sql_cmd_insert_values::execute_inner(THD *thd) {
 
     if (!has_error ||
         thd->get_transaction()->cannot_safely_rollback(Transaction_ctx::STMT)) {
-      if (mysql_bin_log.is_open()) {
+      bool tianmu_engine = insert_table->s->db_type() ? insert_table->s->db_type()->db_type == DB_TYPE_TIANMU: false;
+      if (mysql_bin_log.is_open()&& !tianmu_engine) {
```

Comment:
I think that should be like this: do tianmu check firstly, then binlog flag check. 
if (!tianmu_engine && mysql_bin_log.is_open())